sparkctl/README.md

```diff
@@ -76,11 +77,75 @@ If you want to make uploaded dependencies publicly available so they can be down
 simply add `--public` to the `create` command, as the following example shows:
 
 ```bash
-$ sparkctl create <path to YAML file> --upload-to gs://<bucket> --project <GCP project the GCS bucket is associated to> --public
+$ sparkctl create <path to YAML file> --upload-to gs://<bucket> --public
 ``` 
 
 Publicly available files are referenced through URIs of the form `https://storage.googleapis.com/bucket/path/to/file`.
 
+##### Uploading to S3
+
+For uploading to S3, the value should be in the form of `s3://<bucket>`. The bucket must exist and uploading fails if
+otherwise. The local dependencies will be uploaded to the path
+`spark-app-dependencies/<SparkApplication namespace>/<SparkApplication name>` in the given bucket. It replaces the
+file path of each local dependency with the URI of the remote copy in the parsed `SparkApplication` object if uploading
+is successful.
+
+Note that uploading to S3 with [AWS SDK](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html) 
+requires credentials to be specified. For GCP, the S3 Interoperability credentials can be retrieved as 
+described [here](https://cloud.google.com/storage/docs/migrating#keys).
+SDK uses the default credential provider chain to find AWS credentials. 
+The SDK uses the first provider in the chain that returns credentials without an error.
+The default provider chain looks for credentials in the following order:
+
+- Environment variables
+    ```
+    AWS_ACCESS_KEY_ID
+    AWS_SECRET_ACCESS_KEY
+    ```
+- Shared credentials file (.aws/credentials)
+
+For more information about AWS SDK authentication, please check
+[Specifying Credentials](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials).
+
+Usage:
+```bash
+$ export AWS_ACCESS_KEY_ID=[KEY]
+$ export AWS_SECRET_ACCESS_KEY=[SECRET]
+$ sparkctl create <path to YAML file> --upload-to s3://<bucket>
+```
+
+By default, the uploaded dependencies are not made publicly accessible and are referenced using URIs in the form of
+`s3a://bucket/path/to/file`. To download the dependencies from S3, a custom-built Spark Docker image with the required
+jars for `S3A Connector` (`hadoop-aws-2.7.6.jar`, `aws-java-sdk-1.7.6.jar` for Spark build with Hadoop2.7 profile, or `hadoop-aws-3.1.0.jar`, `aws-java-sdk-bundle-1.11.271.jar` for Hadoop3.1) need to be available in the classpath,
+and `spark-default.conf` with the AWS keys and the S3A FileSystemClass needs to be set (you can also use `spec.hadoopConf` in the SparkApplication YAML):
+
+```
+spark.hadoop.fs.s3a.endpoint https://storage.googleapis.com
+spark.hadoop.fs.s3a.access.key [KEY]
+spark.hadoop.fs.s3a.secret.key [SECRET]
+spark.hadoop.fs.s3a.impl org.apache.hadoop.fs.s3a.S3AFileSystem
+```
+
+NOTE: In Spark 2.3.0 init-containers are used for Kubernetes as resource manager. In future versions, init-containers are removed. 
```

Comment:
`for Kubernetes as resource manager` -> `for downloading remote application dependencies`. Also this applies to any Spark 2.3.x versions.

Reply:
Done